src/_common/modal/modal.ts

```diff
@@ -109,7 +109,7 @@ export default class AppModal extends Vue {
 	}
 
 	dismiss() {
-		this.emitClickAway();
+		this.emitClose();
```

Comment:
Also emit on esc key and you are gucci.

Reply:
I believe it currently is, but only when `esc` is intended to close the modal.
```
dismissEsc() {
	if (this.modal.noEscClose) {
		return;
	}
	this.dismiss();
}
```
Currently `dismissRouteChange()`, `dismissEsc()`, and `dismissBackdrop()` all close the modal by triggering the `dismiss()` function.

Reply:
Ah right, I thought it called resolve.